components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/workflow/SubscriptionDeletionSimpleWorkflowExecutor.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.wso2.carbon.apimgt.impl.workflow;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.apimgt.api.APIManagementException;
+import org.wso2.carbon.apimgt.api.model.APIIdentifier;
+import org.wso2.carbon.apimgt.impl.dao.ApiMgtDAO;
+import org.wso2.carbon.apimgt.impl.dto.SubscriptionWorkflowDTO;
+import org.wso2.carbon.apimgt.impl.dto.WorkflowDTO;
+import org.wso2.carbon.apimgt.impl.utils.APIMgtDBUtil;
+
+import java.sql.Connection;
+import java.sql.SQLException;
+import java.util.List;
+
+/**
+ * Simple workflow executor for subscription delete action
+ */
+public class SubscriptionDeletionSimpleWorkflowExecutor extends WorkflowExecutor {
+
+    private static final Log log = LogFactory.getLog(SubscriptionDeletionSimpleWorkflowExecutor.class);
+
+    @Override
+    public String getWorkflowType() {
+        return WorkflowConstants.WF_TYPE_AM_SUBSCRIPTION_DELETION;
+    }
+
+    @Override
+    public List<WorkflowDTO> getWorkflowDetails(String workflowStatus) throws WorkflowException {
+        return null;
+    }
+
+    @Override
+    public void execute(WorkflowDTO workflowDTO) throws WorkflowException {
+        workflowDTO.setStatus(WorkflowStatus.APPROVED);
+        complete(workflowDTO);
+        super.publishEvents(workflowDTO);
+    }
+
+    @Override
+    public void complete(WorkflowDTO workflowDTO) throws WorkflowException {
+        ApiMgtDAO apiMgtDAO = new ApiMgtDAO();
+        SubscriptionWorkflowDTO subWorkflowDTO = (SubscriptionWorkflowDTO) workflowDTO;
+        Connection conn = null;
+        String errorMsg = null;
+
+        try {
+            APIIdentifier identifier = new APIIdentifier(subWorkflowDTO.getApiProvider(),
+                    subWorkflowDTO.getApiName(), subWorkflowDTO.getApiVersion());
+            int applicationIdID = apiMgtDAO.getApplicationId(subWorkflowDTO.getApplicationName(), subWorkflowDTO
+                    .getSubscriber());
+
+            conn = APIMgtDBUtil.getConnection();
+            conn.setAutoCommit(false);
+            apiMgtDAO.removeSubscription(identifier, applicationIdID, conn);
+            conn.commit();
```

Comment:
I believe what Thilini is suggesting is to move the following logic to the DAO layer and call it from here.

 int applicationIdID = apiMgtDAO.getApplicationId(subWorkflowDTO.getApplicationName(), subWorkflowDTO
 +                    .getSubscriber());
 +
 +            conn = APIMgtDBUtil.getConnection();
 +            conn.setAutoCommit(false);
 +            apiMgtDAO.removeSubscription(identifier, applicationIdID, conn);
 +            conn.commit();